sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/implementation/InputStreamWithReadTimeout.java

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.core.http.jdk.httpclient.implementation;
+
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.SharedExecutorService;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.http.HttpTimeoutException;
+import java.time.Duration;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * An {@link InputStream} that wraps an existing {@link InputStream} and provides read operations with a timeout.
+ */
+public final class InputStreamWithReadTimeout extends InputStream {
+    private final Duration readTimeout;
+    private final boolean hasTimeout;
+    private final InputStream delegate;
+
+    /**
+     * Creates an {@link InputStream} that wraps an existing {@link InputStream} and provides read operations with a
+     * timeout.
+     *
+     * @param delegate The {@link InputStream} to wrap.
+     * @param readTimeout The read timeout.
+     */
+    public InputStreamWithReadTimeout(InputStream delegate, Duration readTimeout) {
+        this.delegate = delegate;
+        this.readTimeout = readTimeout;
+        this.hasTimeout = !readTimeout.isNegative() && !readTimeout.isZero();
+    }
+
+    @Override
+    public int read() throws IOException {
```

Comment:
There are a few new `read*` methods added since Java 9. If this input stream is returned to the user, which we do in some cases, then it will not honor the timeout if the user is using > Java 9 and calls methods like `inputStream.readAllBytes()`.

Reply:
Also, since JdkHttpClient is expected to be used with only JDK 11+, we should override the other read methods as well:
- readNBytes overloads
- readAllBytes
- transferTo